hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeRpcServer.java

```diff
@@ -1650,7 +1650,7 @@ public DatanodeCommand blockReport(final DatanodeRegistration nodeReg,
           final int index = r;
           noStaleStorages = bm.runBlockOp(() ->
             bm.processReport(nodeReg, reports[index].getStorage(),
-                blocks, context));
+                blocks, context, reports.length, index + 1));
```

Comment:
Thanks for reminding me of the shortcomings of my plan. I will try to improve it. However, the solution in this PR may not work. If datanode send one storage report per RPC,  `reports.length` will be 1 here. Your code `totalReportNum == currentReportNum` will always be true. So block report lease will be removed as before. This repairing will be ineffective.

Reply:
Great catch. +1, We have to solve this case.

Reply:
Thanks for reminding me. I think we'll have to think of something else